pandapower/control/controller/trafo/ContinuousTapControl.py

```diff
@@ -43,62 +44,69 @@ def __init__(self, net, tid, vm_set_pu, tol=1e-3, side="lv", trafotype="2W", in_
 
     def control_step(self, net):
         """
         Implements one step of the ContinuousTapControl
         """
-        delta_vm_pu = net.res_bus.at[self.controlled_bus, "vm_pu"] - self.vm_set_pu
+        if self.nothing_to_do(net):
+            return
+
+        delta_vm_pu = read_from_net(net, "res_bus", self.controlled_bus, 'vm_pu', self._read_write_flag) - self.vm_set_pu
         tc = delta_vm_pu / self.tap_step_percent * 100 / self.t_nom
-        self.tap_pos += tc * self.tap_side_coeff * self.tap_sign
+        self.tap_pos = self.tap_pos + tc * self.tap_side_coeff * self.tap_sign
         if self.check_tap_bounds:
             self.tap_pos = np.clip(self.tap_pos, self.tap_min, self.tap_max)
 
         # WRITE TO NET
+        # necessary in case the dtype of the column is int
         if net[self.trafotable].tap_pos.dtype != "float":
             net[self.trafotable].tap_pos = net[self.trafotable].tap_pos.astype(float)
-        net[self.trafotable].at[self.tid, "tap_pos"] = self.tap_pos
+        write_to_net(net, self.trafotable, self.controlled_tid, "tap_pos", self.tap_pos, self._read_write_flag)
 
     def is_converged(self, net):
         """
         The ContinuousTapControl is converged, when the difference of the voltage between control steps is smaller
         than the Tolerance (tol).
         """
-
-        if not net[self.trafotable].at[self.tid, 'in_service']:
+        if self.nothing_to_do(net):
             return True
-        vm_pu = net.res_bus.at[self.controlled_bus, "vm_pu"]
-        self.tap_pos = net[self.trafotable].at[self.tid, 'tap_pos']
+
+        vm_pu = read_from_net(net, "res_bus", self.controlled_bus, "vm_pu", self._read_write_flag)
+        self.tap_pos = read_from_net(net, self.trafotable, self.controlled_tid, "tap_pos", self._read_write_flag)
         difference = 1 - self.vm_set_pu / vm_pu
 
         if self.check_tap_bounds:
-            if self.tap_side_coeff * self.tap_sign == 1:
-                if vm_pu < self.vm_set_pu and self.tap_pos == self.tap_min:
-                    return True
-                elif vm_pu > self.vm_set_pu and self.tap_pos == self.tap_max:
-                    return True
-            elif self.tap_side_coeff * self.tap_sign == -1:
-                if vm_pu > self.vm_set_pu and self.tap_pos == self.tap_min:
-                    return True
-                elif vm_pu < self.vm_set_pu and self.tap_pos == self.tap_max:
-                    return True
-        return abs(difference) < self.tol
+            reached_limit = np.where(self.tap_side_coeff * self.tap_sign == 1,
+                                     (vm_pu < self.vm_set_pu) & (self.tap_pos == self.tap_min) |
+                                     (vm_pu > self.vm_set_pu) & (self.tap_pos == self.tap_max),
+                                     (vm_pu < self.vm_set_pu) & (self.tap_pos == self.tap_max) |
+                                     (vm_pu > self.vm_set_pu) & (self.tap_pos == self.tap_min))
```

Comment:
I don't really understand this part. What is supposed to happen here? Could you add a line of comment on what the result (i.e. reached_limit) should look like?